front/src/common/RollingStockSelector/RollingStockModal.tsx

```diff
@@ -48,97 +107,56 @@ function RollingStockModal(props) {
       }),
     }
   );
+  const [filteredRollingStockList, setFilteredRollingStockList] = useState<LightRollingStock[]>(
+    () => filterRollingStocks(rollingStocks, filters)
+  );
 
-  const searchMateriel = (e) => {
-    setFilters({ ...filters, text: e.target.value.toLowerCase() });
-    setIsFiltering(true);
-  };
+  function handleRollingStockLoaded() {
+    const newFilteredRollingStock = filterRollingStocks(rollingStocks, filters);
+    setFilteredRollingStockList(newFilteredRollingStock);
+  }
 
   const updateSearch = () => {
-    setOpenedRollingStockCardId(undefined);
-
-    const filterText = filters.text;
-    const filterIsNotEmpty = filterText || filters.elec || filters.thermal;
-
-    const filteredRollingStocks = filterIsNotEmpty
-      ? rollingStocks.filter((rollingStock) => {
-          // checkbox filters
-          if (filters.elec || filters.thermal) {
-            const effortCurveModes = Object.values(rollingStock.effort_curves.modes).map(
-              // eslint-disable-next-line camelcase
-              ({ is_electric }) => is_electric
-            );
-            const isElectric = effortCurveModes.includes(true);
-            const isThermal = effortCurveModes.includes(false);
-            if ((filters.elec && !isElectric) || (filters.thermal && !isThermal)) {
-              return false;
-            }
-          }
-
-          // text filter
-          if (filterText) {
-            const { metadata } = rollingStock;
-            const containsFilterText = some(
-              [
-                rollingStock.name,
-                metadata.detail,
-                metadata.reference,
-                metadata.series,
-                metadata.type,
-                metadata.grouping,
-              ],
-              (string) => string && string.toLowerCase().includes(filterText)
-            );
-            if (!containsFilterText) {
-              return false;
-            }
-          }
-
-          // the rolling stock has passed both the checkbox filter and the text filter
-          return true;
-        })
-      : rollingStocks;
+    if (filters.text !== '' || filters.elec !== false || filters.thermal !== false) {
+      setOpenRollingStockCardId(undefined);
+    } else {
+      setOpenRollingStockCardId(rollingStockID);
+    }
+    const newFilteredRollingStock = filterRollingStocks(rollingStocks, filters);
+    if (newFilteredRollingStock) {
```

Comment:
```suggestion
```
No need for the test here as newFilteredRollingStock is never undefined (by default, it will be `[]`)